iterators task (1)/student/include/impl.h

```diff
@@ -2,20 +2,77 @@
 #include <iterator>
 
 template <class Container>
-struct accumulator
+struct accumulator 
 {
-    /** @todo Iterator traits*/
-    /** @todo Constructor with container*/
-    /** @todo Copy assignment operator*/
-    /** @todo Move assignment operator*/
+private:
+    Container& mContainer;
+public:
+    /** Iterator traits*/
+    using iterator_category = std::output_iterator_tag;
+    using value_type =  typename Container::value_type;
+    using difference_type = typename Container::difference_type;
+    using pointer = typename Container::pointer;
+    using reference = typename Container::reference;
+    
+    /** Constructor with container*/
+    explicit accumulator(Container & container)
+    : mContainer (container)
+    {
 
-    /** @todo Iterator operations */
+    }
+
+    /** Copy assignment operator*/  
+    accumulator& operator=(const value_type& value)
+    {
+	    mContainer.push_back(value);
+	    return *this;
+    }
+
+    template <typename T>
+    accumulator& operator=(const T& value)
+    {
+        for (auto && item: value)
+        {
+            mContainer.push_back(item);
+        }
+        return *this;
+    }
+
+    /** Move assignment operator*/
+    accumulator& operator=(value_type&& value)
+    {
+        mContainer.push_back(std::move(value));
+        return *this;
+    }
+
+    template <typename T>
+    accumulator& operator=(T&& value)
```

Comment:
please use more descriptive name, what is value?

Reply:
also, what is T?

Reply:
Agree. Changed.